src/components/parsero/parsero.tsx

```diff
@@ -1,111 +1,177 @@
+     * @param {FormValuesType} values - The form values, containing the URL.
+     */
     const onSubmit = async (values: FormValues) => {
-        // Disallow saving until the tool's execution is complete
-        setAllowSave(false);
-
-        // Enable the Loading Overlay
-        setLoading(true);
+        setAllowSave(false); // Disallow saving until the tool's execution is complete.
+        setLoading(true); // Set the loading state to true to indicate that the process is starting.
 
         const args = [`-u`, values.url];
 
         // Execute parsero
         CommandHelper.runCommandGetPidAndOutput("parsero", args, handleProcessData, handleProcessTermination)
             .then(({ pid, output }) => {
+                // Update the state with the process ID and initial output.
                 setPid(pid);
                 setOutput(output);
             })
             .catch((error) => {
                 setLoading(false);
-                setOutput(`Error: ${error.message}`);
+                setOutput(`Error: ${error.message}`); // If an error occurs during command execution, display the error message.
             });
     };
 
+    /**
+     * clearOutput: Callback function to clear the console output.
+     * It resets the state variable holding the output, thereby clearing the display.
+     */
     const clearOutput = useCallback(() => {
         setOutput("");
         setHasSaved(false);
         setAllowSave(false);
     }, [setOutput]);
 
     return (
-        <form onSubmit={form.onSubmit(onSubmit)}>
-            {LoadingOverlayAndCancelButton(loading, pid)}
-            <Stack>
-                {UserGuide(title, description)}
-                <TextInput label={"url"} required {...form.getInputProps("url")} />
-                <Button type={"submit"}>Start parsero</Button>
-                {SaveOutputToTextFile_v2(output, allowSave, hasSaved, handleSaveComplete)}
-                <ConsoleWrapper output={output} clearOutputCallback={clearOutput} />
-            </Stack>
-        </form>
+        // Render the UserGuide component with component details
+        <RenderComponent
+            title={title}
+            description={description}
+            steps={steps}
+            tutorial={tutorial}
+            sourceLink={sourceLink}
+        >
+            {!loadingModal && (
+                <InstallationModal
+                    isOpen={opened}
+                    setOpened={setOpened}
+                    feature_description={description}
+                    dependencies={dependencies}
+                ></InstallationModal>
+            )}
+            <form onSubmit={form.onSubmit(onSubmit)}>
+                {/* Render the loading overlay and cancel button */}
+                {LoadingOverlayAndCancelButton(loading, pid)}
+                <Stack>
+                    {UserGuide(title, description)}
+                    <TextInput label={"url"} required {...form.getInputProps("url")} />
```

Comment:
Should be URL not url. 
```suggestion
                    <TextInput label={"URL"} required {...form.getInputProps("url")} />
```